soc/xtensa/intel_adsp/common/multiprocessing.c

```diff
@@ -96,6 +96,24 @@ __asm__(".section .text.z_soc_mp_asm_entry, \"x\" \n\t"
 #undef NOP32
 #undef NOP4
 
+__asm__(".align 4\n\t"
+	".global dsp_restore_vector\n\t"
+	"dsp_restore_vector:\n\t"
+	"  movi  a0, 0\n\t"
+	"  movi  a1, 1\n\t"
+	"  movi  a2, 0x40020\n\t"/* PS_UM|PS_WOE */
```

Comment:
Note that once call0 lands, we need a way to select whether or not WOE is set (though I guess in a pinch we could do that in the C entry function -- this is a benign noop as long as code is call0-ABI-pure, as nothing will try to issue an ENTRY or RETW instruction).

Reply:
We are here using xtensa feature with register windows. After core context restore instruction `retw` will result with jump back to the place that lead us tlow power state (pm subsystem in idle thread).